testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,285 @@
+import java.util.stream.Collectors;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertQueryFailure;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryExecutor.query;
+import static io.trino.tests.product.TestGroups.AVRO;
+import static io.trino.tests.product.utils.QueryExecutors.onHive;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public abstract class BaseTestAvroSchemaEvolution
+        extends ProductTest
+{
+    // TODO move Avro schema files to classpath and use tempto SshClient to upload them
+    public static final String ORIGINAL_SCHEMA = "file:///docker/presto-product-tests/avro/original_schema.avsc";
+    public static final String RENAMED_COLUMN_SCHEMA = "file:///docker/presto-product-tests/avro/rename_column_schema.avsc";
+    public static final String REMOVED_COLUMN_SCHEMA = "file:///docker/presto-product-tests/avro/remove_column_schema.avsc";
+    public static final String ADDED_COLUMN_SCHEMA = "file:///docker/presto-product-tests/avro/add_column_schema.avsc";
+    public static final String CHANGE_COLUMN_TYPE_SCHEMA = "file:///docker/presto-product-tests/avro/change_column_type_schema.avsc";
+    public static final String INCOMPATIBLE_TYPE_SCHEMA = "file:///docker/presto-product-tests/avro/incompatible_type_schema.avsc";
+
+    private final String tableName;
+    private final String columnsInTableStatement;
+    private final String selectStarStatement;
+    private final List<Partition> partitions;
+
+    protected BaseTestAvroSchemaEvolution(String tableName, Partition... partitions)
+    {
+        this.tableName = tableName;
+        this.columnsInTableStatement = "SHOW COLUMNS IN " + tableName;
+        this.selectStarStatement = "SELECT * FROM " + tableName;
+        this.partitions = ImmutableList.copyOf(partitions);
+    }
+
+    @BeforeTestWithContext
+    public void createTestTable()
+    {
+        query(format(
+                "CREATE TABLE %s (" +
+                        "  dummy_col VARCHAR" +
+                        (partitions.isEmpty() ? "" : ", " + getPartitionsAsListString(p -> p.getColumnDeclaration())) +
+                        ")" +
+                        "WITH (" +
+                        "  format='AVRO', " +
+                        "  avro_schema_url='%s'" +
+                        (partitions.isEmpty() ? "" : ", partitioned_by=ARRAY[" + getPartitionsAsListString(p -> "'" + p.getName() + "'") + "]") +
+                        ")",
+                tableName,
+                ORIGINAL_SCHEMA));
+        insertData(tableName, 0, "'stringA0'", "0");
+    }
+
+    @AfterTestWithContext
+    public void dropTestTable()
+    {
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+    }
+
+    @Test(groups = AVRO)
+    public void testSelectTable()
```

Comment:
what does the test have to do with schema evolution?

Reply:
It's preexisting test. My understanding is that It verifies that current schema is overwrite by avro_schema_url and not real schema stored in Hive. There is alos test that unsets avro_schema_url and checks if we fallback to oryginal schema.